### ISSUE TITLE
It is not possible to call the smart contract method with the send of money.

### ISSUE THREAD

**Issue by abdulgalimov** [polkadot-js/api#5602]
* **I'm submitting a ...**

  - [x] Bug report
  - [ ] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

When trying to send money to a smart contract:
```
api.rpc.eth.call({
  from: ethAddress,
  to: contractAddress,
  data: encodeAbi,
  value: 100
})
```

I get an error:
```
RPC-CORE: call(request: EthCallRequest, number?: BlockNumber): Bytes:: -32602: invalid type: integer `100`, expected a (both 0x-prefixed or not) hex string or byte array containing between (0; 32] bytes at line 1 column 141
```

If I pass a `value: "0x100",` I also get an error:
```
RPC-CORE: call(request: EthCallRequest, number?: BlockNumber): Bytes:: -32602: invalid type: integer `256`, expected a (both 0x-prefixed or not) hex string or byte array containing between (0; 32] bytes at line 1 column 141
```

* **Please tell us about your environment:**

  - Version: 10.2.1
  - Environment:

    - [x] Node.js
    - [ ] Browser
    - [ ] Other (limited support for other environments)

  - Language:

    - [x] JavaScript
    - [ ] TypeScript (include tsc --version)
    - [ ] Other

**Comment by jacogr** [comment]
In all cases, the `-32602` error code is from the Node itself.

For RPC values, you need to pass _valid_ hex values, not numbers - well, it would obviously depend on _how_ the RPC is created, i.e. some of them do accept numbers as well as hex. Since you are using the RPC directly, ensure that you encode it correctly.

`0x100` is not valid hex, whereas `0x1000` would be. (Hex numbers, when used directly, needs to be in BE format with the correct number of bits as per the type)

**Comment by abdulgalimov** [comment]
When I try to pass "0x0100" I get an error:
```
RPC-CORE: call(request: EthCallRequest, number?: BlockNumber): Bytes:: -32602: invalid type: integer `256`, expected a (both 0x-prefixed or not) hex string or byte array containing between (0; 32] bytes at line 1 column 141
```

The problem is, polkadot.js transforms passed string to number, as it is defined as U256 in embedded rpc definitions.

**Comment by pasynkov** [comment]
@jacogr I have the same prolem with several options of hex: 

ex: `0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000016345785d8a0000`

Could you provide the working approach?

**Comment by abdulgalimov** [comment]
my problem was solved after updating polkadot/api to version 10.7.2
